tests/span_tests.cpp

```diff
@@ -298,70 +297,14 @@ TEST(span_test, from_array_constructor)
 
     int arr2d[2][3] = {1, 2, 3, 4, 5, 6};
 
-#ifdef CONFIRM_COMPILATION_ERRORS
-    {
-        span<int, 6> s{arr};
-    }
-
-    {
-        span<int, 0> s{arr};
-        EXPECT_TRUE(s.size() == 0);
-        EXPECT_TRUE(s.data() == &arr[0]);
```

Comment:
These checks don't make sense when the variable isn't even there because of a compilation error.

Reply:
I wonder if the original author of `CONFIRM_COMPILATION_ERRORS` meant "This doesn't compile, confirm whether that is correct." rather than "This should not compile". Unfortunately, we have no way of determining. Maybe comment it out and see if any older compilers (or C++14) protest?

That being said - Line 303 shouldn't compile because the static extent exceeds the backing array. All the other look fine, I think.

Reply:
What exactly do you mean with
> Maybe comment it out and see if any older compilers (or C++14) protest?

`#ifdef CONFIRM_COMPILATION_ERRORS` => `#if 1`?

All variable declarations in this `CONFIRM_COMPILATION_ERRORS` block fail. With the new `CtorCompilesFor` checks we are sure that the variable definitions are invalid on all platforms. So from my understanding on no single platform the `EXPECT_TRUE(...)` test statements can compile.

Reply:
> What exactly do you mean with <...>

Yeah, `#if 1` is in spirit of what I meant - I just wanted to enable the body to see what fails.

> So from my understanding on no single platform the EXPECT_TRUE(...) test statements can compile.

Do any of these examples compile when using `std::span` instead of `gsl::span`? If so, I'd like to understand why. Otherwise, let's delete the tests.

Reply:
I just tested in a small C++20 project: none of the variable declarations compiles.
So as of right now we do check the declarations don't compile, but we remove the `EXPECT_TRUE()` that could never have been effective.
```
#include <span>

int main() {
    int arr[5] = { 1, 2, 3, 4, 5 };
    {
        std::span<int, 6> s{ arr };
    }

    {
        std::span<int, 0> s{ arr };
    }

    int arr2d[2][3] = { 1, 2, 3, 4, 5, 6 };
    {
        std::span<int> s{ arr2d };
    }

    {
        std::span<int, 0> s{ arr2d };
    }

    {
        std::span<int, 6> s{ arr2d };
    }
    return 0;
}
```